### DISCUSSION TITLE
[Feature] Docusign feature parity

### DISCUSSION THREAD

**Discussion by mparksGP** [docusealco/docuseal#40] [Ideas]
Hello,

My organization recently evaluated DocuSeal for potential use in our operations. We really appreciate this free and open source tool and think it has great potential. However, after careful consideration, we decided not to adopt it at this time due to some missing features that are critical for our use case.

We wanted to provide feedback on the key features we need that would make DocuSeal a viable option for us in the future, with the hope that some day it may be a full-fledged alternative to Docusign:

----

- [ ]  **Signing workflows** 

Currently, DocuSeal sends out notifications to all signers simultaneously. At least for us this is a problem since many documents require counter-signatures or witnesses, and it either does not make sense or is legally impermissible, for later signers to have access to a document before earlier signers are done with it. It would be useful if it was possible to withhold notifications/access to later signers until earlier ones have completed.

- [ ] **Completion notifications**

Only the creator of the template and its signers receive email notifications, and these are sent only when invited and when signing has been completed by each party. It would be useful to:

- Have a separate, configurable address aside from the template owner to be notified when the entire document is finished.
- Have the ability to configure when the email notifications are sent (when done, for some parties only, etc.)

- [ ] **Forgery protection**

One of our compliance people raised this: it is far too easy, accidentally or otherwise, to forge a signature on the behalf of someone else since the share links for any given document are presented upfront in the UI. Some miscellaneous ideas along these lines might be:

- A configurable mode that requires all interaction to go through email (that is, form creators do not have access to sign documents they create)
- Capture more metadata in the digital signature, including the IP address, browser, geolocation, etc. of the signer
- Require separate email confirmation before committing the signature (that is, if I use joe@example.com's signing link, it is impossible to sign the document without him knowing about it)
- Attach a summary of signatures and their metadata directly to the document as another page

I understand that none of these are foolproof: and that this is an inherent risk of self-hosting a document signing platform, but right now it is far too easy to casually do the wrong thing.

- [ ] **Better UI for busy templates**

One set of documents our organization uses has the potential to be signed hundreds of times per month. At this volume, the submissions view would quickly become cluttered and hard to use. A more traditional and denser table view and some kind of search functionality would increase usability under heavy load.

- [ ] **Expiry dates**

Some documents' signatures are only good for a limited time. After this time, there is currently no way for the same person to sign the same document again without manually finding and deleting their submission from the template page. It would be useful to set an optional expiry date for any submission after which the same person could sign again.

- [ ] **Reminders** 

Docusign has the ability to send reminder/nag messages to people who have been invited to sign a document but have not done so in a configurable amount of time, or on demand.. This feature gets a surprising amount of use!

----

 Please let us know if any clarification would be helpful. We're happy to provide additional context on our use case and requirements. Thank you for your efforts on this valuable project!

**Comment by AlexBTurchyn** [comment]
@mparksGP Amazing work. Everything you listed really should be added. Thanks
